ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -158,37 +183,110 @@ func (rdwr *YAMLReadWriter) Get(clusterConfig string, profileConfig string) (*CL
+	if err != nil {
+		return errors.Wrap(err, "Error saving config file")
 	}
 
-	path := configFilePath(rdwr.destination)
+	destMode := rdwr.destination.Mode
+	if err = os.MkdirAll(rdwr.destination.Path, *destMode); err != nil {
+		return err
+	}
 
 	// If config file exists, set permissions first, because we may be writing creds.
-	// This is necessary because ioutil.WriteFile only sets the permissions
-	// if the file is being created for the first time; this handles the case
-	// where the file already exists
 	if _, err := os.Stat(path); err == nil {
 		if err = os.Chmod(path, configFileMode); err != nil {
-			return errors.Wrapf(err, "Could not set file permissions, %s, for path %s", configFileMode, path)
+			logrus.Errorf("Unable to chmod %s to mode %s", path, configFileMode)
+			return err
 		}
 	}
 
-	data, err := yaml.Marshal(cliConfig)
-	if err != nil {
-		return errors.Wrap(err, "Unable to parse the config")
-	}
-	// WriteFile will not change the permissions of an existing file
 	if err = ioutil.WriteFile(path, data, configFileMode.Perm()); err != nil {
-		return errors.Wrapf(err, "Could not write config file %s", path)
+		logrus.Errorf("Unable to write config to %s", path)
+		return err
 	}
 
 	return nil
 }
 
+// SaveProfile saves a single credential configuration
+func (rdwr *YAMLReadWriter) SaveProfile(configName string, profile *Profile) error {
+	path := credentialsFilePath(rdwr.destination)
+
+	config := &ProfileConfig{Profiles: make(map[string]Profile), Version: configVersion}
+	if _, err := os.Stat(path); err == nil {
+		// an existing config file is there
+		config, err = readCredFile(path)
+		if err != nil {
+			return err
+		}
+	}
+
+	config.Profiles[configName] = *profile
+	if len(config.Profiles) == 1 {
+		config.Default = configName
+	}
+
+	// save the modified config
+	return rdwr.saveConfig(path, config)
+}
+
+// SaveCluster save a single cluster configuration
+func (rdwr *YAMLReadWriter) SaveCluster(configName string, cluster *Cluster) error {
+	path := configFilePath(rdwr.destination)
+
+	// if no err on read- then existing yaml config
+	config, err := readClusterFile(path)
+	if err != nil {
+		// err on read: this means that no yaml file currently exists
+		config = &ClusterConfig{Clusters: make(map[string]Cluster), Version: configVersion}
+	}
+
+	config.Clusters[configName] = *cluster
+	if len(config.Clusters) == 1 {
+		config.Default = configName
+	}
+
+	// save the modified config
+	return rdwr.saveConfig(path, config)
+}
+
+// SetDefaultProfile updates which set of credentials is defined as default
+func (rdwr *YAMLReadWriter) SetDefaultProfile(configName string) error {
+	path := credentialsFilePath(rdwr.destination)
+	config, err := readCredFile(path)
+	if err != nil {
+		return err
+	}
+
+	if _, ok := config.Profiles[configName]; ok {
+		config.Default = configName
+	} else {
+		return fmt.Errorf("%s must be defined as a profile before it can be set as default. ", configName)
+	}
+
```

Comment:
Can we swap this so you just return the error on `!ok` in line 263 and then get rid of the else block and have `config.Default = configName` outside of the condition?